src/Symfony/Component/HttpFoundation/RequestMatcher.php

```diff
@@ -76,13 +76,13 @@ public function __construct($path = null, $host = null, $methods = null, $ips =
      */
     public function matchScheme($scheme)
     {
-        $this->schemes = array_map('strtolower', (array) $scheme);
+        $this->schemes = null !== $scheme ? array_map('strtolower', (array) $scheme) : array();
```

Comment:
Not really needed though. Confused with `(array) null` actually converting to an empty array, not something like `array(null)` :)